### PR TITLE
Spike - switching over to --no-gui & --gui

### PR DESCRIPTION
**Pull request by sfkleach**
This is an exploration of the way in which we can switch from --use-build to --no-gui, --gui. N.B. The base branch is not main but the part1 branch. If/when we complete this pull-request we will fold it back into the part1 branch and carry on with the review process there.

### CODE REVIEW

File: makePoplogCommander.sh [hunk 1]
```diff
@@ -828,20 +848,27 @@ cat << \****
...
+            deque_pop_front( argd );
             return processArgs( argd, base, flags, envv );
-        } else if ( startsWith( arg0, "--use-build" ) ) {
-            char * build = strchr( arg0, '=' ) + 1;
+        } else if ( startsWith( arg0, "--gui=" ) ) {
             if ( 0 ) {
                 // Never taken - a trick to regularise the following cases.
 ****
-for variant in "${VARIANTS[@]}"
+for n in {0..2}
```
Comment (willprice) [inline makePoplogCommander.sh]: I'd replace the hard coded upper bound like so:
```bash
for ((i=0; i<${#VARIANT_OPTIONS[@]}; i++)) do
   blah
done
```

File: makePoplogCommander.sh [hunk 2]
```diff
@@ -12,7 +12,15 @@ set -euo pipefail
...
 ################################################################################
 
-VARIANTS=(nox xt xm)
+# The options are what the user types. They have to be translated into builds.
+# e.g. --gui=xt has to become xt. The reason for that is that the 'editions'
+# of Poplog relate to the mutually exclusive build options. The mapping is 
+# implicitly defined by matching position in the two (congruent) arrays.
+VARIANT_OPTIONS=(--no-gui --gui=xt --gui=motif)
+VARIANT_BUILDS=(nox xt xm)
+
```
Comment (willprice) [inline makePoplogCommander.sh:23]: Maybe we should add a little assertion here to make sure these are the same length as a sanity check?
e.g. `[[ ${#VARIANT_OPTIONS[@]} -eq ${#VARIANT_BUILDS[@]} ]] || { echo "VARIANT_OPTIONS should have the same number of items as VARIANT_BUILDS but did not"; exit 1; }`

**Review by willprice — APPROVED** [review]
Nicely done! Just a few minor points. I'd definitely fix the hard coded upper bound, and I suggest adding in the assertion to check the two arrays are the same length (though I'll leave this as optional for merging)

File: makePoplogCommander.sh (hunk 2 above)
Comment (sfkleach) [reply makePoplogCommander.sh:23]: Although it is overkill, it serves as a way of saying 'these two must be the same length" so I have just adopted it. Thanks.

**Pull request merged**